src/main/java/com/epam/lab/paymentsystem/utility/impl/CurrentUserImpl.java

```diff
@@ -0,0 +1,24 @@
+package com.epam.lab.paymentsystem.utility.impl;
+
+import com.epam.lab.paymentsystem.utility.CurrentUser;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Component;
+
+/**
+ * Util class for retrieving information about current logged user.
+ */
+@Component
+public class CurrentUserImpl implements CurrentUser {
```

Comment:
If you use `@Component` then you have to `@Autowire` it as you did. But this component naming is not clear. When it is called `CurrentUser` I expect it to be static util class or smth like that but not the autowirable component. Please, rename `CurrentUser` and `CurrentUserImpl` to something more suitable